File: projects/G4R8AJ/C#/ProgNyelvekBead/States/UnsignedStart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProgNyelvekBead.States
{
    class UnsignedStart : AbstractState
    {
        public UnsignedStart(String word, String curVal) : base(word, curVal) { }

        protected override void calculateNextState(Char curChar)
        {
            if(curChar == '.')
            {
                this.remWord = this.remWord.Remove(0, 1);
                this.curVal = "0.";
                this.followingState = new PointState(this.remWord,this.curVal);
                
            }
            else if(curChar == '0')
            {
                this.remWord = this.remWord.Remove(0, 1);
                this.curVal = "0";
                this.followingState = new ZeroState(this.remWord, this.curVal); 
            }
            else if( Char.IsDigit(curChar))
            {
                this.followingState = new IntegralState(this.remWord, this.curVal);
            }
            else
            {
                this.followingState = null;
            }

            this.isStepped = true;
        }
    }
}

File: projects/G4R8AJ/C++/src/UnsignedStart.cpp
#include "UnsignedStart.h"

UnsignedStart::UnsignedStart()
{
    this->isTerminal = false;
}

AbstractState::State UnsignedStart::next(std::string str,std::string curVal)
{

    State ret;
    ret.currentVal = curVal;
    if(str[0] ==  '.')
    {
        PointState nextState;
        ret.nextState = new PointState();
        ret.currentVal = curVal + "0.";
        ret.currentRemaining = str.erase(0,1);
    }
    else if( str[0] == '0' )
    {
        ret.nextState = new ZeroState();
        ret.currentVal = curVal + "0";
        ret.currentRemaining = str.erase(0,1);
    }
    else if( isdigit(str[0]) )
    {
        ret.nextState = new IntegralState();
        ret.currentVal = curVal + str[0];
        ret.currentRemaining = str.erase(0,1);
    }
    else
    {
        ret.nextState = NULL;
    }
    return ret;
}
